src/relay/backend/aot_executor_codegen.cc

```diff
@@ -587,8 +633,12 @@ class AOTExecutorCodegen : public MixedModeVisitor {
     dict_attrs.Set("global_symbol", run_func_name);
     dict_attrs.Set("runner_function", Bool(true));
 
+    tir::Stmt device_activations = GenerateAllDeviceHook("Activate");
+    tir::Stmt device_deactivations = GenerateAllDeviceHook("Deactivate");
+    tir::Stmt final_body = tir::SeqStmt({device_activations, body, device_deactivations});
```

Comment:
Should we just re-use body here ?

Reply:
My personal opinion is we should create and name variables for their purpose rather than constantly mutating them so you can track what they're being used for. Shouldn't the C++ compiler figure out that `body` is no longer live and do this micro-optimisation for us?